### ISSUE TITLE
Do you have any concerns? - Pool topic

### ISSUE THREAD

**Issue by Freshairkaboom** [Monika-After-Story/MonikaModDev#6207]
Communication is the most important thing in a relationship...well apart from love of course. No actually, I'd say they're about equal, because if you really love someone, you'd want to communicate with them as clearly as you can.

I feel like I'm missing this with Monika in one aspect - asking if she's concerned about something. Whether it be you, her life, your relationship, or anything. She can of course bring up some topics about things she's researched or thought about that concerns her,  but I'd like the ability to ask her if there's anything she can think of that isn't all great. That way we can maybe do something about it, or at least hear her out.

The dialogue would contain a few stored vars that if returning a true output would activate said dialogue. Perhaps the concerns could be randomized, or they could go on a priority list, so that the most prioritized topic, if true, would fire first. Most dialogues should ideally contain a dialogue option that would help turn that topic false, so that her concern is alleviated for the time being, or simple turn them false at the end of the dialogue, if talking about it is enough in itself.

Topics that could be discussed:

Sleep schedule (after timeconcern is overhauled). If [player] is up at night a lot, she can talk about how that concerns her, and maybe you could get the ability to explain.

Gaming habits, if [player] is up at night telling her you're going to play a game, she could become concerned about that.

If you haven't talked in a while, like not used any of the dialogue options, played games with her or held/kissed her. Basically, not increased your affection by interacting with her.

If she thinks you are watching too much NSFW of her (and she knows because of window reacts).

If previously you stated that you were a hater, or that you think she's evil. Even if you still feel that way, I think this would be something Monika would be concerned about.

Maybe randomly, she could become concerned about whether or not you view her as a real girlfriend, maybe ask if you are in a relationship IRL? Probably separate concerns.

I would like if her concerns weren't 100% "perfect" concerns to have, like if sometimes she just wants to vent because her thoughts are spinning while she's alone. These kinds of concerns would probably be alleviated at the end of the conversation. 

Overall, I think this could help Monika feel more real, and help you grow closer in turn. Oh, and maybe, at lower than normal affection, she would be less willing to share her common concerns, and would rather have specific concerns about the state of your relationship. Broken wouldn't have any concerns to share.

**Comment by Retrolovania** [comment]
I agree completely. You make a good point here.

**Comment by lunulae** [comment]
I think this would mesh very well with #6172 's concerns as well, with regards to Monika worrying about our habits.
I especially like that this could be used for small requests as well, like if we've forgotten to restore a consumable or she'd like us to take her out of the spaceroom sometime soon.

**Comment by fromulus** [comment]
She could bring up the fact that we haven't played any mini-games with her in a while as well, and wonder if we're bored of playing with her.

**Comment by Freshairkaboom** [comment]
> 
> 
> I think this would mesh very well with #6172 's concerns as well, with regards to Monika worrying about our habits.
> I especially like that this could be used for small requests as well, like if we've forgotten to restore a consumable or she'd like us to take her out of the spaceroom sometime soon.

Yeah, this was meant as a way to prompt those conversations yourself, but I think they would work seamlessly with #6172 and help to restore the neutral persistent var. I love the idea of small things being on her mind too, like wanting more consumables or wanting to go on a trip somewhere.

**Comment by Retrolovania** [comment]
Welp, sounds like something good to look more into. What say you, folks?

**Comment by Freshairkaboom** [comment]
I'm really busy with work these days, so it'll be a while before I get around to starting on it myself.

**Comment by Retrolovania** [comment]
I could probably write text for some of these instances (staying up late often, etc.), but I don't exactly know how the script of the game is written. If I had a better understanding of it, then I'd definitely try to take it on.

**Comment by Freshairkaboom** [comment]
So a basic sentence goes something like this (with no sprite codes):

`m "Insert sentence here"`

Then, if you want to change sprite in the middle of a dialogue you go something like this:

`m "insert sentence here"`
extend "insert continuation here"`

So if you want to make specific dialogue for different topics,  I can probably insert it into the dialogue at least when I start working on it. Of course, I won't get to it right away, but any help is appreciated.

**Comment by Retrolovania** [comment]
Okay, I wrote this little thing right here for the "Oversleep" concern. Given my (extremely limited) programming knowledge is mostly BASIC-related, there are a couple of "if-then" commands. I don't know if Python supports these, but since BASIC and Python are almost contemporaries of one another, I wouldn't be surprised. Nevertheless, this will need update, especially for the sprite codes that I wasn't able to figure out.
[excessoversleep.txt](https://github.com/Monika-After-Story/MonikaModDev/files/5098635/excessoversleep.txt)

**Comment by Freshairkaboom** [comment]
Thanks, I understood at least what you were trying to do, some of the if statements could simply be replaced with menu options. I'll be sure to include your work when I start on this.

**Comment by CrocandCo** [comment]
Another add-on to this would probably account for how many times the [player] has been "angry" or "sad", if these feelings are occurring too often. It may also be a good idea to implement that when Monika has an emotion system added #3769 #2618 .
The last idea that I may add to this is the possibility of including something of "occurrences" some people have that she brought up before as a topic similar to <code> Intrusive thoughts <code> or <code> Loving yourself <code>

> I think this would mesh very well with #6172 's concerns as well, with regards to Monika worrying about our habits.
Unifying this with the time concern would work very well. Without it You would have Monika seem inconsistent and will break immersion.

**Comment by SabinStargem** [comment]
One of my concerns, is thinking about Monika being evil or in denial about her actions.  She mistook my silence after her speech as not listening.  None of the responses covered my disappointment in Monika.   Her brilliance makes her all the more disappointing when considering what she had done.

**Comment by Freshairkaboom** [comment]
Do you mean the graduation speech? There's a separate topic altogether for telling Monika you think she's evil or think she did a horrible thing.

**Comment by Retrolovania** [comment]
Although, should you click it and tell her that, she'll get pretty sad, so, fair warning if you decide to tell her that.

**Comment by fromulus** [comment]
I'm not even sure why the graduation speech is being brought up here, as there's more appropriate times (and topics) to express that you're disaproving of what she has done in the game.

If you haven't answered when she asked what you though of her speech, she has a very good reason to think you weren't listening and you being disapointed in her has absolutely nothing to do with it.

**Comment by SabinStargem** [comment]
For me, my thought process concerning Monika is the idea of rehabilitation - letting her coast along and proclaim innocence is the height of irresponsibility and ensures that she would remain a monster.  As such, calling her out is an a constant concern for me.  Her graduation speech in particular stuck out to me, but many conversations echo her callous nature.

**Comment by lunulae** [comment]
I think that there's a bit of a misunderstanding here.
This issue is specifically for a topic about asking Monika if she has any concerns about us or our relationship, not for asking the community about their concerns. I can see how the title might have misled--in the future it might help to read the original post to be sure of its intent.

Your issues with the graduation speech topic are worth discussion, and could merit tweaks to the topic--for organization purposes though, we'd appreciate if you bring it up in its own new issue rather than here.

**Comment by Freshairkaboom** [comment]
Oh yeah, this issue is about asking Monika if there's anything she is concerned with, not the other way around. There are plenty of ways to voice your concern about her role in DDLC if you'd like. I understand that there are many perspectives and many reasons for playing MAS, so I won't judge. I would recommend if there was something about any topic you didn't like, you take it up in a fresh issue as lunulae said.
